### PR TITLE
Add ServiceHub helpers recommended in previous PR

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Remote/ServiceHub/ExternalAccess/Pythia/Api/PythiaBrokeredServiceImplementation.cs [hunk 1]
```diff
@@ -18,7 +18,11 @@ public static ValueTask<T> RunServiceAsync<T>(Func<CancellationToken, ValueTask<
         public static ValueTask RunServiceAsync(Func<CancellationToken, ValueTask> implementation, CancellationToken cancellationToken)
             => BrokeredServiceBase.RunServiceImplAsync(implementation, cancellationToken);
 
+        [Obsolete("Use RunWithSolutionAsync instead", error: false)]
         public static ValueTask<Solution> GetSolutionAsync(this PythiaPinnedSolutionInfoWrapper solutionInfo, ServiceBrokerClient client, CancellationToken cancellationToken)
             => RemoteWorkspaceManager.Default.GetSolutionAsync(client, solutionInfo.UnderlyingObject, cancellationToken);
+
+        public static ValueTask<T> RunWithSolutionAsync<T>(this PythiaPinnedSolutionInfoWrapper solutionInfo, ServiceBrokerClient client, Func<Solution, ValueTask<T>> implementation, CancellationToken cancellationToken)
```
Comment (tmat) [inline src/Workspaces/Remote/ServiceHub/ExternalAccess/Pythia/Api/PythiaBrokeredServiceImplementation.cs:25]: Can these also be overloads of RunServiceAsync?

Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/ServiceHub/ExternalAccess/Pythia/Api/PythiaBrokeredServiceImplementation.cs:25]: sure

Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/ServiceHub/ExternalAccess/Pythia/Api/PythiaBrokeredServiceImplementation.cs]: done.

**Comment by JoeRobich** [comment]
Retargeting to main so we can remove main-vs-deps branch

File: src/Workspaces/Remote/ServiceHub/ExternalAccess/Pythia/Api/PythiaBrokeredServiceImplementation.cs [hunk 2]
```diff
@@ -18,7 +18,11 @@ public static ValueTask<T> RunServiceAsync<T>(Func<CancellationToken, ValueTask<
         public static ValueTask RunServiceAsync(Func<CancellationToken, ValueTask> implementation, CancellationToken cancellationToken)
             => BrokeredServiceBase.RunServiceImplAsync(implementation, cancellationToken);
 
+        [Obsolete("Use RunWithSolutionAsync instead", error: false)]
```
Comment (davidwengier) [inline src/Workspaces/Remote/ServiceHub/ExternalAccess/Pythia/Api/PythiaBrokeredServiceImplementation.cs:21]: Message needs updating here, and everywhere else

**Review by davidwengier — APPROVED** [review]

**Pull request merged**